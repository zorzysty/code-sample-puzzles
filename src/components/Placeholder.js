import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { DropTarget } from 'react-dnd';


function collect(connect, monitor) {
  return {
    connectDropTarget: connect.dropTarget(),
    hovered: monitor.isOver(),
  };
}

const spec = {
  drop: (props, monitor, component) => ({
    // set success prop of dropResult depending on whether piece was placed in correct place
    success: monitor.getItem().id === component.props.index,
  }),
};

class Placeholder extends PureComponent {
  render() {
    const { connectDropTarget, hovered } = this.props;

    return connectDropTarget(
      <div className={`Placeholder ${hovered ? 'Placeholder-hovered' : ''}`} />,
    );
  }
}

Placeholder.propTypes = {
  connectDropTarget: PropTypes.func.isRequired,
  hovered: PropTypes.bool.isRequired,
  index: PropTypes.number.isRequired, // eslint-disable-line react/no-unused-prop-types
};

export default DropTarget('piece', spec, collect)(Placeholder);
