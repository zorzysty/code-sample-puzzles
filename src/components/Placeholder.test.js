import React from 'react';
import { shallow } from 'enzyme';
import Placeholder from './Placeholder';


describe('Placeholder', () => {
  it('should render correctly based on passed props', () => {
    const propsArray = [
      {
        key: 0,
        index: 0,
      },
      {
        key: 6,
        index: 6,
      },
    ];

    propsArray.forEach((props) => {
      const component = shallow(<Placeholder {...props} />);

      expect(component)
        .toMatchSnapshot();
    });
  });
});
