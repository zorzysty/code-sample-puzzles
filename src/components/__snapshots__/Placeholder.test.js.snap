// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder should render correctly based on passed props 1`] = `ShallowWrapper {}`;

exports[`Placeholder should render correctly based on passed props 2`] = `ShallowWrapper {}`;
